db/migrate/20170529142557_unify_container_definition_and_container.rb

```diff
@@ -0,0 +1,169 @@
+    has_one :container, :class_name => 'UnifyContainerDefinitionAndContainer::Container'
+  end
+
+  class Container < ActiveRecord::Base
+    belongs_to :container_definition, :class_name => 'UnifyContainerDefinitionAndContainer::ContainerDefinition'
+  end
+
+  class SecurityContext < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+    belongs_to :resource, :polymorphic => true
+  end
+
+  class ContainerEnvVar < ActiveRecord::Base
+    belongs_to :container_definition, :class_name => 'UnifyContainerDefinitionAndContainer::ContainerDefinition'
+  end
+
+  class ContainerPortConfig < ActiveRecord::Base
+    belongs_to :container_definition, :class_name => 'UnifyContainerDefinitionAndContainer::ContainerDefinition'
+  end
+
+  def up
+    # attributes
+    add_column :containers, :image,              :string
+    add_column :containers, :image_pull_policy,  :string
+    add_column :containers, :memory,             :string
+    add_column :containers, :cpu_cores,          :float
+    add_column :containers, :container_group_id, :bigint
+    add_column :containers, :privileged,         :boolean
+    add_column :containers, :run_as_user,        :bigint
+    add_column :containers, :run_as_non_root,    :boolean
+    add_column :containers, :capabilities_add,   :string
+    add_column :containers, :capabilities_drop,  :string
+    add_column :containers, :command,            :text
+
+    containers = Arel::Table.new(:containers)
+    definitions = Arel::Table.new(:container_definitions)
+    say_with_time("Copying over columns from container_definition to container") do
+      %w(image image_pull_policy memory cpu_cores container_group_id privileged
+         run_as_user run_as_non_root capabilities_add capabilities_drop command).each do |column|
+        join_sql = definitions.project(definitions[column.to_sym])
+                              .where(definitions[:id].eq(containers[:container_definition_id])).to_sql
+        Container.update_all("#{column} = (#{join_sql})")
+      end
+    end
+
+    say_with_time("switch container_definition_id with container_id for container_port_configs") do
+      port_configs = Arel::Table.new(:container_port_configs)
+      join_sql = containers.project(containers[:id])
+                           .where(containers[:container_definition_id].eq(port_configs[:container_definition_id])).to_sql
+      ContainerPortConfig.update_all("container_definition_id = (#{join_sql})")
+    end
+
+    say_with_time("switch container_definition_id with container_id for container_port_configs") do
+      env_vars = Arel::Table.new(:container_env_vars)
+      join_sql = containers.project(containers[:id])
+                           .where(containers[:container_definition_id].eq(env_vars[:container_definition_id])).to_sql
+      ContainerEnvVar.update_all("container_definition_id = (#{join_sql})")
+    end
+
+    say_with_time("switch container_definition_id with container_id for security_contexts") do
```

Comment:
swtich resource_id with container_id, resource_type to 'Container' for security_contexts